blocks/trending-news/trending-news.js

```diff
@@ -0,0 +1,113 @@
+  block.textContent = '';
+
+  function createNewsItem(news) {
+    const article = document.createElement('div');
+    article.className = 'article';
+
+    const mediaWrapper = document.createElement('div');
+    mediaWrapper.className = 'picture-wrapper';
+
+    const picture = document.createElement('picture');
+    const img = document.createElement('img');
+    img.src = news.imgUrl;
+    img.alt = '';
+    img.loading = 'lazy';
+    picture.appendChild(img);
+    mediaWrapper.appendChild(picture);
+
+    const textContent = document.createElement('div');
+    textContent.className = 'article-text';
+
+    const h3 = document.createElement('h3');
+    h3.className = 'post_title';
+    const a = document.createElement('a');
+    a.href = news.link;
+    a.target = '_blank';
+    a.tabIndex = '0';
+    a.textContent = news.title;
+    h3.appendChild(a);
+    textContent.appendChild(h3);
+
+    const postMeta = document.createElement('div');
+    postMeta.className = 'post-info';
+    const imgTime = document.createElement('img');
+    imgTime.src = 'https://www.icicidirect.com/Content/images/time.svg';
+    imgTime.alt = 'time';
+    imgTime.loading = 'lazy';
+    const abbr = document.createElement('abbr');
+    abbr.textContent = `${news.date} `;
+    const abbrSource = document.createElement('abbr');
+    abbrSource.textContent = news.source;
+    postMeta.appendChild(imgTime);
+    postMeta.appendChild(abbr);
+    postMeta.appendChild(abbrSource);
+    textContent.appendChild(postMeta);
+
+    const description = document.createElement('div');
+    description.className = 'descriptn articleDesc';
+    description.id = `divtrendingStock_${Math.floor(Math.random() * 10000000000)}`;
+    textContent.appendChild(description);
+
+    article.appendChild(mediaWrapper);
+    article.appendChild(textContent);
+
+    return article;
+  }
+
+  const container = document.createElement('div');
+  container.className = 'container';
+
+  const titleWrap = document.createElement('div');
+  titleWrap.className = 'title text-center';
+  const h2 = document.createElement('h2');
+  h2.textContent = 'TRENDING NEWS';
+  titleWrap.appendChild(h2);
+  container.appendChild(titleWrap);
+
+  const newsSection = document.createElement('div');
+  newsSection.className = 'news-section';
+
+  const slider = document.createElement('div');
+  slider.className = 'news-slider';
+
+  const slickListDragable = document.createElement('div');
+  slickListDragable.className = 'news-list';
+
+  const slickTrack = document.createElement('div');
+  slickTrack.className = 'news-track';
+
+  newsData.forEach((news) => {
+    const slide = document.createElement('div');
+    slide.className = 'news-card';
+    const article = createNewsItem(news);
+    slide.appendChild(article);
+    slickTrack.appendChild(slide);
+  });
+
+  slickListDragable.appendChild(slickTrack);
+  slider.appendChild(slickListDragable);
+  newsSection.appendChild(slider);
+
+  // Discover more link
```

Comment:
this part of decoration can be moved into another method. Please see if we can break this decorate function into smaller methods